src/main/java/io/r2dbc/postgresql/PostgresqlSqlLexer.java

```diff
@@ -124,12 +124,12 @@ private static boolean isSpecialOrOperatorChar(char c) {
     private static TokenizedSql.Token getBlockCommentToken(String sql, int beginIndex) {
         int depth = 1;
         for (int i = beginIndex + 2; i < (sql.length() - 1); i++) {
-            String biGraph = sql.substring(i, i + 2);
-
-            if (biGraph.equals("/*")) {
+            char c1 = sql.charAt(i);
+            char c2 = sql.charAt(i + 1);
+            if (c1 == '/' && c2 == '*') {
```

Comment:
Apologies but to me current version seems more readable `biGraph.equals("/*")` i.e. start of the comment block. 

Or am I missing something here?

Reply:
`sql.substring(i, i + 2)`.
Using char instead can avoid generating redundant string objects.